src/lib/annotations/doc/DocDrawingDialog.js

```diff
@@ -1,41 +1,254 @@
+import * as annotatorUtil from '../annotatorUtil';
+import * as constants from '../annotationConstants';
+import { ICON_DRAW_SAVE, ICON_DRAW_DELETE } from '../../icons/icons';
+
+const LABEL_TEMPLATE = `<span class="${constants.CLASS_ANNOTATION_DRAWING_LABEL} ${constants.CLASS_HIDDEN}"></span>`;
+const COMMIT_TEMPLATE = `
+    <button class="bp-btn-plain ${constants.CLASS_ADD_DRAWING_BTN}"
+        title="${__('annotation_draw_save')}">
+        ${ICON_DRAW_SAVE}
+        Save
+    </button>`.trim();
+const DELETE_TEMPLATE = `
+    <button class="bp-btn-plain ${constants.CLASS_DELETE_DRAWING_BTN}"
+        title="${__('annotation_draw_delete')}">
+        ${ICON_DRAW_DELETE}
+        Delete
+    </button>`.trim();
 
 class DocDrawingDialog extends AnnotationDialog {
+    /** @property {boolean} Whether or not the dialog is visible */
+    visible = false;
+
+    /**
+     * [constructor]
+     *
+     * @param {AnnotationDialogData} data - Data for constructing drawing dialog
+     * @return {DocDrawingDialog} Drawing dialog instance
+     */
+    constructor(data) {
+        super(data);
+
+        this.postDrawing = this.postDrawing.bind(this);
+        this.deleteAnnotation = this.deleteAnnotation.bind(this);
+    }
+
     /**
-     * Empty stub to avoid unexpected behavior.
+     * [destructor]
+     *
+     * @return {void}
+     */
+    destroy() {
+        this.unbindDOMListeners();
+        this.removeAllListeners();
+
+        if (!this.element) {
+            return;
+        }
+
+        this.element.removeEventListener('click', annotatorUtil.prevDefAndStopProp);
+        if (this.pageEl && this.pageEl.contains(this.element)) {
+            this.pageEl.removeChild(this.element);
+        }
+    }
+
+    /**
+     * Returns whether or not the dialog is able to be seen
+     *
+     * @public
+     * @return {boolean} Whether or not the dialog is able to be seen
+     */
+    isVisible() {
+        return this.visible;
+    }
+
+    /**
+     * Save the drawing thread upon clicking save. Will cause a soft commit.
      *
      * @override
      * @protected
+     * @return {void}
      */
     addAnnotation() {}
 
     /**
-     * Empty stub to avoid unexpected behavior.
+     * Empty stub to avoid unexpected behavior. Removing a drawing annotation can only be done by deleting the thread.
      *
      * @override
      * @protected
+     * @return {void}
      */
     removeAnnotation() {}
 
     /**
+     * Bind dialog button listeners
+     *
+     * @protected
+     * @return {void}
+     */
+    bindDOMListeners() {
+        if (this.commitButtonEl) {
+            this.commitButtonEl.addEventListener('click', this.postDrawing);
+            this.commitButtonEl.addEventListener('touchend', this.postDrawing);
+        }
+
+        if (this.deleteButtonEl) {
+            this.deleteButtonEl.addEventListener('click', this.deleteAnnotation);
+            this.deleteButtonEl.addEventListener('touchend', this.deleteAnnotation);
```

Comment:
I believe click will fire in the touch chain of things. If not we we should only add the touch listeners if the browser `hasTouch`